stratum/CMakeLists.txt

```diff
@@ -295,55 +295,58 @@ if(WITH_OPENCONFIG)
     target_link_libraries(stratum PUBLIC openconfig_proto)
 endif()
 
+find_library(LIBTDI tdi ${SDE_INSTALL_PREFIX}/lib)
+if(NOT LIBTDI)
+    message(FATAL_ERROR "Cannot find library: tdi")
+endif()
+
+find_library(LIBTDI_JSON_PARSER tdi_json_parser ${SDE_INSTALL_PREFIX}/lib)
+if(NOT LIBTDI_JSON_PARSER)
+    message(FATAL_ERROR "Cannot find library: tdi_json_parser")
+endif()
+
+find_library(LIBTARGET_SYS target_sys ${SDE_INSTALL_PREFIX}/lib)
+if(NOT LIBTARGET_SYS)
+    message(FATAL_ERROR "Cannot find library: target_sys")
+endif()
+
 if(TOFINO_TARGET)
     find_library(LIBDRIVER driver ${SDE_INSTALL_PREFIX}/lib)
     if(NOT LIBDRIVER)
         message(FATAL_ERROR "Cannot find library: driver")
     endif()
 
-    find_library(LIBTDI tdi ${SDE_INSTALL_PREFIX}/lib)
-    if(NOT LIBTDI)
-        message(FATAL_ERROR "Cannot find library: tdi")
-    endif()
-
-    find_library(LIBTDI_JSON_PARSER tdi_json_parser ${SDE_INSTALL_PREFIX}/lib)
-    if(NOT LIBTDI_JSON_PARSER)
-        message(FATAL_ERROR "Cannot find library: tdi_json_parser")
-    endif()
-
-    find_library(LIBTARGET_SYS target_sys ${SDE_INSTALL_PREFIX}/lib)
-    if(NOT LIBTARGET_SYS)
-        message(FATAL_ERROR "Cannot find library: target_sys")
-    endif()
-
     target_link_libraries(stratum PUBLIC
         ${LIBDRIVER}
         ${LIBTDI}
         ${LIBTDI_JSON_PARSER}
         ${LIBTARGET_SYS}
     )
-else()
-    find_library(TARGET_UTILS target_utils ${SDE_INSTALL_PREFIX}/lib)
-    if(NOT TARGET_UTILS)
-        message(FATAL_ERROR "Cannot find library: target_utils")
+elseif(DPDK_TARGET)
+    find_library(LIBDRIVER driver ${SDE_INSTALL_PREFIX}/lib)
```

Comment:
is this a recent change? Before this, DPDK had a requirement for libbf_switchd_lib or something like that?
libdriver was only introduced with cmake in p4studio,

Reply:
We still link against libbf_switchd_lib. It's the item after this one in the CMake file.